src/lib.rs

```diff
@@ -195,6 +203,54 @@ struct NdcBoundingCircle {
     radius: f32,
 }
 
+struct DebugCursor;
+
+/// Updates the 3d cursor to be in the pointed world coordinates
+fn update_debug_cursor_position(
+    pick_state: ResMut<PickState>,
+    mut query: Query<(&DebugCursor, &mut Translation)>,
+    mut camera_query: Query<(&Transform, &Camera)>,
+) {
+    // Get the camera
+    let mut view_matrix = Mat4::zero();
+    let mut projection_matrix = Mat4::zero();
+    for (transform, camera) in &mut camera_query.iter() {
+        view_matrix = transform.value.inverse();
+        projection_matrix = camera.projection_matrix;
+    }
+
+    // Set the cursor translation to the top pick's world coordinates
+    if let Some(top_pick) = pick_state.top() {
+        let pos = top_pick.get_pick_coord_world(projection_matrix, view_matrix);
+
+        for (_, mut translation) in &mut query.iter() {
+            translation.0.set_x(pos.x());
```

Comment:
Can
`translation.0 = pos`
be used instead of passing individual axes?

Reply:
Oops, yeah. Fixed!